internal/kafka/internal/handlers/kafka_test.go

```diff
@@ -0,0 +1,591 @@
+package handlers
+
+import (
+	"bytes"
+	"context"
+	"net/http"
+	"testing"
+
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/internal/kafka/internal/api/dbapi"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/internal/kafka/internal/config"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/internal/kafka/internal/kafkas/types"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/internal/kafka/internal/services"
+	mocks "github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/internal/kafka/test/mocks/kafkas"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/api"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/auth"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/errors"
+	s "github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/services"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/services/authorization"
+	"github.com/golang-jwt/jwt/v4"
+	"github.com/gorilla/mux"
+	. "github.com/onsi/gomega"
+)
+
+var (
+	id  = "test-id"
+	ctx = auth.SetTokenInContext(context.TODO(), &jwt.Token{
+		Claims: jwt.MapClaims{
+			"username":     "test-user",
+			"org_id":       "",
+			"is_org_admin": true,
+		},
+	})
+	limit = 2
+)
+
+func Test_KafkaHandler_Get(t *testing.T) {
+	type fields struct {
+		service        services.KafkaService
+		providerConfig *config.ProviderConfig
+		authService    authorization.Authorization
+		kafkaConfig    *config.KafkaConfig
+	}
+
+	tests := []struct {
+		name           string
+		fields         fields
+		wantStatusCode int
+	}{
+		{
+			name: "should succeed if kafkaService GET succeeds",
+			fields: fields{
+				service: &services.KafkaServiceMock{
+					GetFunc: func(ctx context.Context, id string) (*dbapi.KafkaRequest, *errors.ServiceError) {
+						return mocks.BuildKafkaRequest(mocks.WithPredefinedTestValues()), nil
+					},
+				},
+				kafkaConfig: &FullKafkaConfig,
+			},
+			wantStatusCode: http.StatusOK,
+		},
+		{
+			name: "should fail if kafkaService GET fails",
+			fields: fields{
+				service: &services.KafkaServiceMock{
+					GetFunc: func(ctx context.Context, id string) (*dbapi.KafkaRequest, *errors.ServiceError) {
+						return nil, errors.GeneralError("test")
+					},
+				},
+			},
+			wantStatusCode: http.StatusInternalServerError,
+		},
+	}
+
+	RegisterTestingT(t)
+
+	for _, testcase := range tests {
+		tt := testcase
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			h := NewKafkaHandler(tt.fields.service, tt.fields.providerConfig, tt.fields.authService, tt.fields.kafkaConfig)
+			req, rw := GetHandlerParams("GET", "/{id}", nil)
+			req = mux.SetURLVars(req, map[string]string{"id": id})
+			h.Get(rw, req)
+			resp := rw.Result()
+			Expect(resp.StatusCode).To(Equal(tt.wantStatusCode))
+			resp.Body.Close()
```

Comment:
If the `Expect` would fail, the test would exit without closing the body. Move this before the `Expect`

Reply:
> If the `Expect` would fail, the test would exit without closing the body. Move this before the `Expect`

Thank you @ziccardi ! Just pushed the changes.